From 8b3e71f01de31941c41b09b14aa7eff12b5845f4 Mon Sep 17 00:00:00 2001
From: stheinan <stheinan@maxlinear.com>
Date: Fri, 26 Jul 2024 11:28:01 +0530
Subject: [PATCH] WLANRTSYS-83410 Invalid MLDID sent in remove MLD command.

Issue:
Invalid MLDID is sent in remove mld command message causing the assert
in firmware.Preventive check was added in hostapd to avoid sending invalid
mldid,porting the same to hostap-ng.

Fix:
Avoid sending remove mld command incase mldid is invalid.

FIXUP to WLANRTSYS-60843 MLO changes: AP MLD configuration
---
 src/ap/mxl_mld.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ap/mxl_mld.c b/src/ap/mxl_mld.c
index 51bd6604c..17077da67 100644
--- a/src/ap/mxl_mld.c
+++ b/src/ap/mxl_mld.c
@@ -1334,6 +1334,10 @@ bool mxl_is_ml_probe(const u8 * ml_ie, size_t ml_ie_len)
 int mxl_hostapd_mld_remove(struct hostapd_data *hapd, u8 sendto_fw)
 {
 	struct mxl_mld_remove mld_rem;
+	if (hapd->conf->mld_id == INVALID_MLD_ID) {
+		wpa_printf(MSG_ERROR, "Invalid MLDID while mld remove");
+		return -1;
+	}
 	mld_rem.mld_id = hapd->conf->mld_id;
 	mld_rem.sendto_fw = sendto_fw;
 	wpa_printf(MSG_INFO, "Removing MLD sendto_fw %u", sendto_fw);
-- 
2.43.0

